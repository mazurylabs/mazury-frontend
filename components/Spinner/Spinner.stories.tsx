import { ComponentMeta, ComponentStory } from '@storybook/react';

import { Spinner } from './Spinner';

export default {
  title: 'Components/Spinner',
  component: Spinner,
  argTypes: {
    size: {
      control: {
        type: 'number',
      },
    },
  },
} as ComponentMeta<typeof Spinner>;

const Template: ComponentStory<typeof Spinner> = (args) => (
  <Spinner {...args} />
);

export const Default = Template.bind({});
Default.args = {
  size: 64,
};
