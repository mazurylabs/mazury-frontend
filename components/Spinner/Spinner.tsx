import Image from 'next/image';
import { FC } from 'react';

interface SpinnerProps {
  size?: number | string;
  className?: string;
}

export const Spinner: FC<SpinnerProps> = ({ size, className }) => {
  const sizeValue = size || '64px';

  return (
    <Image
      src="/icons/loader.svg"
      width={sizeValue}
      height={sizeValue}
      alt="Loading indicator"
      className={`mx-auto animate-spin ${className}`}
    />
  );
};
