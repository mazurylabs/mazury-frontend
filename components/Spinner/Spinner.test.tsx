import { render } from '@testing-library/react';
import { Spinner } from './Spinner';

test('Spinner', () => {
  const { container } = render(<Spinner />);
  expect(container).toMatchSnapshot();
});
