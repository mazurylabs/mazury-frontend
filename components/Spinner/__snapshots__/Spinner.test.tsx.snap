// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner 1`] = `
<div>
  <span
    style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
  >
    <span
      style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
    >
      <img
        alt=""
        aria-hidden="true"
        src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNjQiIGhlaWdodD0iNjQiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
        style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
      />
    </span>
    <img
      alt="Loading indicator"
      class="mx-auto animate-spin undefined"
      data-nimg="intrinsic"
      decoding="async"
      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
    />
    <noscript />
  </span>
</div>
`;
